ambry-frontend/src/main/java/com.github.ambry.frontend/ThrottlingHandler.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.frontend;
+
+import com.github.ambry.config.FrontendConfig;
+import com.github.ambry.rest.RestMethod;
+import com.github.ambry.rest.RestRequest;
+import com.github.ambry.utils.RejectThrottler;
+import java.util.HashMap;
+import java.util.Map;
+
+
+/**
+ * A throttlers class to throttle requests based on request context.
+ */
+public class ThrottlingHandler {
```

Comment:
i wonder if we should call this `QuotaManager` :) I think this is the piece of code that will evolve to become the quota manager eventually

Reply:
Cool name.